cmd/readPipelineEnv.go

```diff
@@ -1,42 +1,80 @@
 package cmd
 
 import (
+	"crypto/aes"
+	"crypto/cipher"
+	"crypto/rand"
+	"crypto/sha256"
+	"encoding/base64"
 	"encoding/json"
+	"fmt"
+	"github.com/SAP/jenkins-library/pkg/config"
 	"github.com/SAP/jenkins-library/pkg/log"
+	"github.com/SAP/jenkins-library/pkg/orchestrator"
 	"github.com/SAP/jenkins-library/pkg/piperenv"
 	"github.com/spf13/cobra"
+	"io"
 	"os"
 	"path"
 )
 
 // ReadPipelineEnv reads the commonPipelineEnvironment from disk and outputs it as JSON
 func ReadPipelineEnv() *cobra.Command {
+	var stepConfig artifactPrepareVersionOptions
+	metadata := artifactPrepareVersionMetadata()
+
 	return &cobra.Command{
 		Use:   "readPipelineEnv",
 		Short: "Reads the commonPipelineEnvironment from disk and outputs it as JSON",
 		PreRun: func(cmd *cobra.Command, args []string) {
 			path, _ := os.Getwd()
 			fatalHook := &log.FatalHook{CorrelationID: GeneralConfig.CorrelationID, Path: path}
 			log.RegisterHook(fatalHook)
+
+			err := PrepareConfig(cmd, &metadata, "", &stepConfig, config.OpenPiperFile)
+			if err != nil {
+				log.SetErrorCategory(log.ErrorConfiguration)
+				return
+			}
+			log.RegisterSecret(stepConfig.Password)
+			log.RegisterSecret(stepConfig.Username)
 		},
 
 		Run: func(cmd *cobra.Command, args []string) {
-			err := runReadPipelineEnv()
+			err := runReadPipelineEnv(&stepConfig)
 			if err != nil {
 				log.Entry().Fatalf("error when writing reading Pipeline environment: %v", err)
 			}
 		},
 	}
 }
 
-func runReadPipelineEnv() error {
+func runReadPipelineEnv(config *artifactPrepareVersionOptions) error {
 	cpe := piperenv.CPEMap{}
 
 	err := cpe.LoadFromDisk(path.Join(GeneralConfig.EnvRootPath, "commonPipelineEnvironment"))
 	if err != nil {
 		return err
 	}
 
+	// try to encrypt
+	if config.Password != "" && orchestrator.DetectOrchestrator() == orchestrator.GitHubActions {
```

Comment:
Instead of having the logic orchestrator-dependent in here, I would prefer to add encryption/decryption flags to these steps. In that way, we have more control - it'll be easier to turn encryption on and off. We can then handle the logic from the orchestrator extension, when these steps are called.

This would be especially necessary for Jenkins and Azure later on, because we can't simply enable encryption in Piper for them, as a lot of custom pipelines can break.

I would then also let the step fail if the password is not there, and give a meaningful error message, so that the user knows that the artifactPrepareVersion password needs to be there.